public/app/core/utils/explore.ts

```diff
@@ -249,10 +250,13 @@ export function parseUrlState(initial: string | undefined): ExploreUrlState {
   };
   const datasource = parsed[ParseUrlStateIndex.Datasource];
   const parsedSegments = parsed.slice(ParseUrlStateIndex.SegmentsStart);
-  const queries = parsedSegments.filter((segment) => !isSegment(segment, 'ui', 'originPanelId', 'mode'));
+  const queries = parsedSegments.filter(
+    (segment) => !isSegment(segment, 'ui', 'originPanelId', 'mode', ...preferredVisualizationTypes)
+  );
 
-  const originPanelId = parsedSegments.filter((segment) => isSegment(segment, 'originPanelId'))[0];
-  return { datasource, queries, range, originPanelId };
+  const originPanelId = parsedSegments.find((segment) => isSegment(segment, 'originPanelId'));
+  const panelsState = parsedSegments.find((segment) => isSegment(segment, ...preferredVisualizationTypes));
```

Comment:
Basically, any element in the array that has any of the preferredVisualizationTypes keys is assumed to be panel state. @gabor had mentioned some desire to improve URL handling - I'm just fitting into the existing pattern for now.

Reply:
This is the main reason we want to change the URL pattern before this because this is very brittle. It is easy to break this in various ways. For example, if some query contains `{trace: 'something'}` explore will stop working for that data source, and because the query can be anything we cannot even be sure there aren't third-party data sources that do not have such keys. Obviously, this was a problem before but adding more keys just makes it more likely.

Reply:
Ahh, good call - I had not considered that. 

I can modify this so that panel state is serialized using a key that is very unlikely to have a collision with any datasource. IE, something like `{ __panelState: {trace : {spanId: 'asdf' } } }`

It will take up a few more characters in the url, but it will be safe to avoid breaking datasources until y'all are able to update the URL pattern.

Reply:
Fixed